### DISCUSSION TITLE
Is there a way to create a login portal for your Quarto website so people need permissions to access?

### DISCUSSION THREAD

**Discussion by braunrs** [quarto-dev/quarto-cli#6052] [Q&A]
### Description

I am trying to add privacy settings to my Quarto website to have them create a login and password to enable them to view the content, or at least allow me to create login/passwords to provide them for people within my network to view the website content, rather than having it publicly available.

**Comment by mcanouil** [comment]
Search for <https://duckduckgo.com/?q=htaccess+password>.

Also, see:
- #3669
- #3182

**Reply by braunrs** [reply]
I am writing my Quarto website through R, and it is currently being passed through GitHub, but I want to change that so that I can make users provide a password to access the data that it will contain. I am having trouble finding the proper examples to show either where to post this in the R code (I know the examples you listed showed the _quarto.yml file) or where to host the website to allow for the htaccess and password to apply.

**Reply by mcanouil** [reply]
> I am writing my Quarto website through R, and it is currently being passed through GitHub, but I want to change that so that I can make users provide a password to access the data that it will contain. I am having trouble finding the proper examples to show either where to post this in the R code (I know the examples you listed showed the _quarto.yml file) or where to host the website to allow for the htaccess and password to apply.

I am afraid I don't quite understand what you are saying here or what's your issue ...

If you really want to do everything in R, use `cat()` or `writeLines()` to write the files.

Regarding the where, any web server would do the job, this includes any of the services natively supported by Quarto and described in the Publishing part of the documentation, see <https://quarto.org/docs/publishing/>.

**Comment by fredguth** [comment]
Another idea is to use [PageCrypt](https://github.com/Greenheart/pagecrypt).

**Comment by lukmayer** [comment]
For anyone in the future who might see this, I have made a relatively simple guide for how to achieve something akin to what OP asked for.

[Here is the link to my blog post about how to set up gated content for a quarto site](https://lukmayer.github.io/blog/posts/login.html)

**Reply by mcanouil** [reply]
@lukmayer Please add an actual message and a proper hyperlink text if you decide to hide the URL.
Without action from you we will have to consider this spam.
Thank you for your understanding.

Note that "here", "link to post", or "updated link" are really bad for accessibility.
There is no information provided in these about what the links are about.

**Comment by chriscarrollsmith** [comment]
Deploy to Azure App Service and use their "Easy Auth".

**Comment by lukmayer** [comment]
Hi Mickaël,

Thank you for the heads up.
I have edited my original post to be clearer about the link's contents. 
I hope this is suitable?

Best, 
Luke

**Reply by Eflores89** [reply]
link is broken 😟

**Reply by lukmayer** [reply]
I have updated the link.
